### PR TITLE
refactor: use map for namespace reexports by name

### PR DESCRIPTION
**Pull request by dnalborczyk**
This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### CODE REVIEW

**Comment by dnalborczyk** [comment]
@lukastaegert is this: https://github.com/rollup/rollup/pull/4411/commits/f7434c3453969463d600ca56ed974fc930764888 possibly a left-over from some previous iteration of the code? should this remain in place? I couldn't find any purpose in the synchronous execution flow. if there is one, it's not covered by a test.

**Review by guybedford — APPROVED** [review]

File: src/Module.ts [hunk 1]
```diff
@@ -463,8 +464,6 @@ export default class Module {
 		if (this.transitiveReexports) {
 			return this.transitiveReexports;
 		}
-		// to avoid infinite recursion when using circular `export * from X`
-		this.transitiveReexports = [];
```
Comment (lukastaegert) [inline src/Module.ts:467]: I would very much expect this one to be needed, maybe a test is missing. After all, there is a recursion in line 475 which needs to be broken if there is a cycle. Will need to look into this.

Comment (lukastaegert) [reply src/Module.ts:467]: I pushed a failing test for this one

**Review by lukastaegert — CHANGES_REQUESTED** [review]

File: src/Module.ts (hunk 1 above)
Comment (dnalborczyk) [reply src/Module.ts:467]: awesome! thank you! I'll revert.

**Review by lukastaegert — APPROVED** [review]

**Pull request merged**